xknx/remote_value/remote_value_setpoint_shift.py

```diff
@@ -37,23 +47,40 @@ def __init__(
             after_update_cb=after_update_cb,
         )
 
+        self.dpt_class: type[DPTValue1Count | DPTTemperature] | None = (
+            setpoint_shift_mode.value if setpoint_shift_mode is not None else None
+        )
         self.setpoint_shift_step = setpoint_shift_step
 
     def payload_valid(self, payload: DPTArray | DPTBinary | None) -> DPTArray | None:
         """Test if telegram payload may be parsed."""
-        # pylint: disable=no-self-use
-        return (
-            payload
-            if isinstance(payload, DPTArray) and len(payload.value) == 1
-            else None
-        )
+        if not isinstance(payload, DPTArray):
+            return None
+        payload_length = len(payload.value)
+        if self.dpt_class is None:
+            if payload_length == DPTTemperature.payload_length:
+                self.dpt_class = DPTTemperature
+            elif payload_length == DPTValue1Count.payload_length:
+                self.dpt_class = DPTValue1Count
+            else:
+                return None
```

Comment:
I'd add the old default here or?

Reply:
`payload_valid` is only called for incoming telegrams so here it should be fine. There will be a payload_length in every case, if it is not 1 or 2 its invalid anyway.

We could set the old default (DPT 6.010) in the Climate classes attribute again - it is `None` now, but I'm not specifically happy with this because DPT6.010 also relies setpoint_shift_step to be configured correctly. I'd favor DPT9 as it is independent of other attributes - if we want to set a default. Then again we'd have to explicitly set it to `None` from HA to have automatic mode enabled again. 
A climate device without a setpoint_shift value can't set a new target temperature because `initialized_for_setpoint_shift_calculations()` returns False. So i think the missing default would only be noticed if the value is set from the (recently introduced) RemoteValue.value setter from code directly, otherwise there would have been a telegram with a payload initializing the RV.